indico/modules/events/abstracts/models/reviews.py

```diff
@@ -0,0 +1,230 @@
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from indico.core.db.sqlalchemy import db, PyIntEnum, UTCDateTime
+from indico.util.date_time import now_utc
+from indico.util.string import format_repr, return_ascii
+from indico.util.struct.enum import IndicoEnum
+
+
+class ReviewProposedAction(int, IndicoEnum):
+    accept = 1
+    reject = 2
+    merge = 3
+    mark_as_dupe = 4
+    change_track = 5
+
+
+class AbstractReview(db.Model):
+    """Represents an abstract review, emitted by a reviewer"""
+
+    __tablename__ = 'abstract_reviews'
+    __table_args__ = (db.UniqueConstraint('abstract_id', 'user_id', 'track_id'),
+                      {'schema': 'event_abstracts'})
+
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    abstract_id = db.Column(
+        db.Integer,
+        db.ForeignKey('event_abstracts.abstracts.id'),
+        index=True,
+        nullable=False
+    )
+    user_id = db.Column(
+        db.Integer,
+        db.ForeignKey('users.users.id'),
+        nullable=False,
+        index=True
+    )
+    track_id = db.Column(
+        db.Integer,
+        nullable=True
+    )
+    created_dt = db.Column(
+        UTCDateTime,
+        nullable=False,
+        default=now_utc,
+        index=True
+    )
+    modified_dt = db.Column(
+        UTCDateTime,
+        nullable=True
+    )
+    comment = db.Column(
+        db.Text,
+        nullable=True
+    )
+    proposed_action = db.Column(
+        PyIntEnum(ReviewProposedAction),
+        nullable=False
+    )
+    proposed_track_id = db.Column(
+        db.Integer,
+        nullable=True
+    )
+    proposed_type_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.contribution_types.id'),
+        nullable=True,
+        index=True
+    )
+    abstract = db.relationship(
+        'Abstract',
+        lazy=False,
+        backref=db.backref(
+            'judgments',
+            lazy='dynamic'
+        )
+    )
+    user = db.relationship(
+        'User',
+        lazy=False,
+        backref=db.backref(
+            'abstract_judgments',
+            lazy='dynamic'
+        )
+    )
+    track = db.relationship(
+        'Track',
+        lazy=False,
+        foreign_keys=[track_id],
+        backref=db.backref(
+            'abstract_reviews',
+            lazy='dynamic'
+        )
+    )
+    proposed_track = db.relationship(
+        'Track',
+        lazy=False,
+        foreign_keys=[proposed_track_id],
+        backref=db.backref(
+            'abstract_reviews',
```

Comment:
you use the same backref name in the previous relationship. they need to be inuqie